open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/validate/VRPSubmissionTest.java

```diff
@@ -0,0 +1,761 @@
+import com.wso2.openbanking.accelerator.consent.mgt.service.impl.ConsentCoreServiceImpl;
+import net.minidev.json.JSONObject;
+import net.minidev.json.parser.JSONParser;
+import net.minidev.json.parser.ParseException;
+import org.joda.time.Instant;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.testng.Assert;
+import org.testng.IObjectFactory;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.ObjectFactory;
+import org.testng.annotations.Test;
+
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+
+/**
+ * Test class for validating Variable Recurring Payment submission requests.
+ */
+@PrepareForTest({OpenBankingConfigParser.class, OpenBankingConfigParser.class, ConsentServiceUtil.class})
+@PowerMockIgnore({"com.wso2.openbanking.accelerator.consent.extensions.common.*", "net.minidev.*",
+        "jdk.internal.reflect.*"})
+public class VRPSubmissionTest {
+    VRPSubmissionPayloadValidator validator = new VRPSubmissionPayloadValidator();
+
+    DefaultConsentValidator consentValidator;
+    @Mock
+    ConsentValidateData consentValidateDataMock;
+    @Mock
+    DetailedConsentResource detailedConsentResourceMock;
+    @Mock
+    ConsentCoreServiceImpl consentCoreServiceMock;
+
+    Map<String, String> resourceParams = new HashMap<>();
+    JSONObject headers = new JSONObject();
+    private static Map<String, String> configMap;
+    Map<String, String> consentAttributes = new HashMap<>();
+    ArrayList<AuthorizationResource> authorizationResources = new ArrayList<AuthorizationResource>();
+
+    @BeforeClass
+    public void initClass() throws ReflectiveOperationException {
+        MockitoAnnotations.initMocks(this);
+
+        //to execute util class initialization
+        new CarbonUtils();
+        System.setProperty("some.property", "property.value");
+        System.setProperty("carbon.home", ".");
+        ConsentExtensionTestUtils.injectEnvironmentVariable("CARBON_HOME", ".");
+
+        configMap = new HashMap<>();
+        configMap.put("ErrorURL", "https://localhost:8243/error");
+
+        consentValidator = new DefaultConsentValidator();
+        consentValidateDataMock = mock(ConsentValidateData.class);
+        authorizationResources.add(getAuthorizationResource());
+        detailedConsentResourceMock = mock(DetailedConsentResource.class);
+        consentCoreServiceMock = mock(ConsentCoreServiceImpl.class);
+    }
+
+    @BeforeMethod
+    public void initMethod() {
+
+        OpenBankingConfigParser openBankingUKConfigParserMock =  mock(OpenBankingConfigParser.class);
```

Comment:
Shall we remove UK config parser from accelerator and only use accelerator config parser?

Reply:
`Fixed in [991e96c]